baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/RawTypes.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.baseline.errorprone;
+
+import com.google.auto.service.AutoService;
+import com.google.errorprone.BugPattern;
+import com.google.errorprone.VisitorState;
+import com.google.errorprone.bugpatterns.BugChecker;
+import com.google.errorprone.matchers.Description;
+import com.google.errorprone.util.ASTHelpers;
+import com.sun.source.tree.ClassTree;
+import com.sun.source.tree.MethodTree;
+import com.sun.source.tree.NewClassTree;
+import com.sun.source.tree.ParameterizedTypeTree;
+import com.sun.source.tree.Tree;
+import com.sun.source.tree.VariableTree;
+import com.sun.tools.javac.code.Type;
+import java.util.ArrayList;
+import java.util.List;
+
+@AutoService(BugChecker.class)
+@BugPattern(
+        name = "RawTypes",
+        link = "https://github.com/palantir/gradle-baseline#baseline-error-prone-checks",
+        linkType = BugPattern.LinkType.CUSTOM,
+        severity = BugPattern.SeverityLevel.WARNING,
+        // Support SuppressWarnings("rawtypes"), which is already in use
+        altNames = {"rawtypes"},
+        summary =
+                "Avoid raw types; add appropriate type parameters if possible.\n"
+                        + "This can be suppressed with @SuppressWarnings(\"rawtypes\") where necessary, such as when "
+                        + "interacting with older library code.")
+public final class RawTypes extends BugChecker
+        implements BugChecker.VariableTreeMatcher,
+                BugChecker.NewClassTreeMatcher,
+                BugChecker.ClassTreeMatcher,
+                BugChecker.MethodTreeMatcher {
+    @Override
+    public Description matchVariable(VariableTree tree, VisitorState state) {
+        return testType(tree.getType());
+    }
+
+    @Override
+    public Description matchNewClass(NewClassTree tree, VisitorState state) {
+        return testType(tree.getIdentifier());
+    }
+
+    @Override
+    public Description matchClass(ClassTree tree, VisitorState state) {
+        List<Tree> types = new ArrayList<>();
```

Comment:
nit: we should try to avoid array allocations since this will be invoked for every class/interface/enum/etc. A large internal project will benefit from avoiding this.

Perhaps something along these lines?
```java
checkType(tree.getExtendsClause());
checkTypes(tree.getImplementsClause());
```